salt/cleanup/remove-post-orchestration.sls

```diff
@@ -0,0 +1,32 @@
+include:
+  - kubectl-config
+
+{%- set target = salt['pillar.get']('target') %}
+{%- set nodename = salt.caasp_net.get_nodename(host=target) %}
+
+###############
+# k8s cluster
+###############
+
+{%- set k8s_nodes = salt['mine.get']('roles:(kube-master|kube-minion)', 'nodename', tgt_type='grain_pcre').keys() %}
+{%- if target in k8s_nodes %}
+
+{%- from '_macros/kubectl.jinja' import kubectl with context %}
+
+{{ kubectl("remove-node",
+           "delete node " + nodename) }}
+
+{% endif %}
+
+###############
+# etcd node
+###############
+
+{%- set etcd_members = salt['mine.get']('roles:etcd', 'nodename', tgt_type='grain').keys() %}
+{%- if target in etcd_members %}
+
+etcd-remove-member:
+  caasp_etcd.member_remove:
+  - nodename: {{ nodename }}
+
+{%- endif %}
```

Comment:
I would remove both conditionals, because the regular removal orchestration could have deleted mine information about the target and failed afterwards, so:

```
caasp-admin:~ # docker exec -it 8d8 salt-run mine.get roles:etcd fun=nodename tgt_type=grain
51a0b22209fd41888e94e1c5c7e226bc:
    caasp-master-1
b0b2b64f0c0940d3ba6a244a342df1d4:
    caasp-master-0
cc64dbdeab4f4e12a75356da132bfc96:
    caasp-master-2
15:37:17,686 [INFO    ] Runner completed: 20180420153712662936
caasp-admin:~ # docker exec -it 8d8 salt-run cache.clear_all 51a0b22209fd41888e94e1c5c7e226bc
True
15:38:28,026 [INFO    ] Runner completed: 20180420153822977102
caasp-admin:~ # docker exec -it 8d8 salt-run mine.get roles:etcd fun=nodename tgt_type=grain
b0b2b64f0c0940d3ba6a244a342df1d4:
    caasp-master-0
cc64dbdeab4f4e12a75356da132bfc96:
    caasp-master-2
15:38:35,590 [INFO    ] Runner completed: 20180420153830618428
```

Since the `salt.state` on the orchestration has `fail_minions` it's fine if any of those fail (and since they don't have requirements for each other it's also fine). So, something like this:

```saltstack
include:
  - kubectl-config

{%- set target = salt['pillar.get']('target') %}
{%- set nodename = salt.caasp_net.get_nodename(host=target) %}

###############
# k8s cluster
###############

{%- from '_macros/kubectl.jinja' import kubectl with context %}

{{ kubectl("remove-node",
           "delete node " + nodename) }}

###############
# etcd node
###############

etcd-remove-member:
  caasp_etcd.member_remove:
  - nodename: {{ nodename }}
```

Reply:
Added a `forced` argument, as discused on IRC.

Reply:
Trying again now.